src/include/ipc4/module.h

```diff
@@ -66,6 +66,13 @@ union ipc4_extended_param_id {
 	} part;
 } __attribute__((packed, aligned(4)));
 
+struct vendor_error {
+	/* Index of the failed parameter */
+	uint32_t failed_param_index;
+	/* Error code */
+	uint32_t vendor_error_code;
```

Comment:
This struct is already called vendor, I think using just `err_code` is fine. Also, you can just use `param_index` because we already know the context from struct name. 

```

struct ipc4_vendor_error {
uint32_t param_idx;
uint32_t err_code
};

```